torch/_inductor/codegen/triton.py

```diff
@@ -756,10 +756,9 @@ def maybe_upcast_float32(convert_output: bool = True):
     """
 
     def needs_upcast(var) -> bool:
-        return (
-            not config.triton.codegen_upcast_to_fp32
-            and isinstance(var, CSEVariable)
-            and var.dtype in (torch.float16, torch.bfloat16)
+        return isinstance(var, CSEVariable) and var.dtype in (
```

Comment:
did you remove the config?

Reply:
cc @eellison

Reply:
This is fine. Previously it was gated because there was no situation we expected to see bf16 vars but there's no reason we need the gate. In the general case we should be upcasting before libdevice ops which dont support bf16.

Reply:
I actually ended up reverting this change because it is actually *also* used for downcasting.

Reply:
I don't follow , would you say more ? For ops which dont support low precision, it should upcast, run op, then downcast after.